src/bvals/comms/coalesced_buffers.cpp

```diff
@@ -0,0 +1,482 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#include <cstdio>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "basic_types.hpp"
+#include "bvals/comms/bnd_id.hpp"
+#include "bvals/comms/bvals_utils.hpp"
+#include "bvals/comms/coalesced_buffers.hpp"
+#include "bvals/neighbor_block.hpp"
+#include "coordinates/coordinates.hpp"
+#include "interface/mesh_data.hpp"
+#include "interface/variable.hpp"
+#include "mesh/mesh.hpp"
+#include "mesh/meshblock.hpp"
+#include "utils/communication_buffer.hpp"
+
+namespace parthenon {
+
+//----------------------------------------------------------------------------------------
+void CoalescedBuffer::AllocateCoalescedBuffer() {
+  int send_rank = sender ? Globals::my_rank : other_rank;
+  int recv_rank = sender ? other_rank : Globals::my_rank;
+  coalesced_comm_buffer = CommBuffer<buf_t>(
+      2 * partition, send_rank, recv_rank, comm_,
+      [](int size) { return buf_t("Combined Buffer", 2 * size); }, true);
+
+  sparse_status_buffer =
+      CommBuffer<std::vector<int>>(2 * partition + 1, send_rank, recv_rank, comm_);
+  sparse_status_buffer.ConstructBuffer(current_size + 1);
```

Comment:
General question regarding sparse: AthenaPK is still build by default with `PARTHENON_DISABLE_SPARSE=ON`. Are there any implication wrt the sparse machinery of the coalesced buffers?

Reply:
In principle this should work with `PARTHENON_DISABLE_SPARSE=On`, but I haven't tested that.